CReLU.lua

```diff
@@ -0,0 +1,51 @@
+local CReLU, parent = torch.class('nn.CReLU', 'nn.Sequential')
+
+-- Implements the CReLU activation function as described by
+-- W. Shang et al. in "Understanding and Improving Convolutional Neural Networks
+-- via Concatenated Rectified Linear Units"
+-- Note: Only supports batched input of the form [B,C,H,W]
+function CReLU:__init(inplace)
+   parent.__init(self)
+   local concatTable = nn.ConcatTable()
+   concatTable:add(nn.Identity())
+   concatTable:add(nn.MulConstant(-1))
+   self.inplace = inplace or false
+   self.batched = true
+   self:add(concatTable)
+   self:add(nn.JoinTable(2))
+   self:add(nn.ReLU(self.inplace))
+end
+
+
+function CReLU:updateOutput(input)
+   self.batched = true
+   if input:nDimension() == 3 then
+      self.batched = false
+   end
+
+   if self.batched then
+      parent.updateOutput(self, input)
+   else
+      parent.updateOutput(self, input:view(1, input:size(1), input:size(2), input:size(3)))
```

Comment:
@soumith I had a look at the use cases of `nInputDims`, I'm a bit confused about how it should be used with `CReLU`. For example, if I want to perform `CReLU` over a 4-dim tensor (batched) I would initialize the module with `nn.CReLU(4)`. But if I want to do it over a 3-dim tensor (non-batched), would I use `nn.CReLU(3)`, if so in the implementation above for an input of size `[3,4,5]` the output would be of size `[3,8,5]` not the expected `[6,4,5]`

I think I'm just misunderstanding how `nInputDims` should be interpreted